src/test/shell/bazel/remote/remote_execution_test.sh

```diff
@@ -3232,44 +3232,73 @@ EOF
     --verbose_failures \
     //a:consumer >& $TEST_log || fail "Failed to populate the cache"
 
-  bazel clean || fail "Failed to clean"
+  bazel clean >& $TEST_log || fail "Failed to clean"
 
-  # download top level outputs without remote metadata
+  # download top level outputs
   bazel build \
     --remote_cache=grpc://localhost:${worker_port} \
     --remote_download_toplevel \
     --verbose_failures \
-    //a:consumer >& $TEST_log || fail "Failed to download outputs without remote metadata"
-  (! [[ -f bazel-bin/a/a.txt ]] && ! [[ -f bazel-bin/a/b.txt ]]) \
-  || fail "Expected outputs of producer are not downloaded without remote metadata"
+    //a:consumer >& $TEST_log || fail "Failed to download outputs"
+  [[ -f bazel-bin/a/a.txt ]] || [[ -f bazel-bin/a/b.txt ]] \
+    && fail "Expected outputs of producer are not downloaded"
 
-  # build without remote cache without remote metadata
+  # build without remote cache
   echo 'bar' > a/in.txt
   bazel build \
     --remote_download_toplevel \
     --verbose_failures \
-    //a:consumer >& $TEST_log || fail "Failed to build without remote cache without remote metadata"
+    //a:consumer >& $TEST_log || fail "Failed to build without remote cache"
+}
 
-  bazel clean || fail "Failed to clean"
+function test_download_toplevel_when_turn_remote_cache_off_with_metadata() {
```

Comment:
These 2 test cases differ by 1 option only -- why not put the test case as a separate helper function and call that twice with different parameters (Bash tests parameterization). Example [here](https://github.com/bazelbuild/bazel/blob/bb2e0ae6021da27cb4d6044bcb5c26e00d2b3db3/src/test/shell/integration/client_sigint_test.sh#L80-L86).

Reply:
Done.